base/common/DateLUTImpl.h

```diff
@@ -243,37 +272,45 @@ class DateLUTImpl
 
     inline time_t toTime(time_t t) const
     {
-        DayNum index = findIndex(t);
+        auto index = findIndex(t);
 
         if (unlikely(index == 0))
             return t + offset_at_start_of_epoch;
 
         time_t res = t - lut[index].date;
 
-        if (res >= lut[index].time_at_offset_change)
-            res += lut[index].amount_of_offset_change;
+        if (res >= lut[index].time_at_offset_change * 900)
+            res += lut[index].amount_of_offset_change * 900;
 
         return res - offset_at_start_of_epoch; /// Starting at 1970-01-01 00:00:00 local time.
     }
 
     inline unsigned toHour(time_t t) const
     {
-        DayNum index = findIndex(t);
+        auto index = findIndex(t);
 
         /// If it is not 1970 year (findIndex found nothing appropriate),
         ///  than limit number of hours to avoid insane results like 1970-01-01 89:28:15
-        if (unlikely(index == 0))
-            return static_cast<unsigned>((t + offset_at_start_of_epoch) / 3600) % 24;
+//        if (unlikely(index == 0))
+//            return static_cast<unsigned>((t + offset_at_start_of_epoch) / 3600) % 24;
 
         time_t res = t - lut[index].date;
 
         /// Data is cleaned to avoid possibility of underflow.
-        if (res >= lut[index].time_at_offset_change)
-            res += lut[index].amount_of_offset_change;
+        if (res >= lut[index].time_at_offset_change * 900)
```

Comment:
That is a minimal timezone offset to reduce field size from `Int32` to `Int8`, in order to keep `Value` size the same, while having full `time_t` value rather than just low 32bits of it.